src/main/java/seedu/address/model/module/Module.java

```diff
@@ -114,6 +115,21 @@ public Module addAssignment(Assignment assignment) {
         }
     }
 
+    /**
+     * Adds a grade to the GradeTracker of the module.
+     *
+     * @param grade grade to add to grade tracker.
+     * @return Module a new module with the grade added.
+     */
+    public Module addGrade(Grade grade) {
+        if (Grade.isValidGrade(grade.gradeResult)) {
+            gradeTracker.setGrade(grade);
+            return new Module(name, zoomLink, gradeTracker);
+        } else {
+            return this;
+        }
+    }
```

Comment:
I thought we decided on using an association class instead? Im not too sure also. KIV